docs/orchestrating-elastic-stack-applications/maps.asciidoc

```diff
@@ -0,0 +1,214 @@
+
+. Download the basemap zip archive using the link shown in the {ems} UI or extracted from the `/status` endpoint.
+. Create a PersistentVolumeClaim of sufficient size (> 90G) and a temporary Pod to mount the corresponding volume
++
+[source,yaml]
+----
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: ems-basemap
+spec:
+  storageClassName: "standard"
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 250G
+---
+kind: Pod
+apiVersion: v1
+metadata:
+  name: ems-data-setup
+spec:
+  terminationGracePeriodSeconds: 0
+  volumes:
+    - name: ems-storage
+      persistentVolumeClaim:
+       claimName: ems-basemap
+  containers:
+    - name: ems-setup
+      image: ubuntu
+      command: [bash, -c, "apt-get update && apt-get install unzip && while true; do sleep 10; done"]
+      volumeMounts:
+        - mountPath: "/usr/share/planet"
+          name: ems-storage
+----
+. Use `kubectl` to copy the basemap data into the volume
++
+[source,sh]
+----
+kubectl cp planet.zip ems-data-setup:/usr/share/planet/planet.zip
+----
+. Unzip the archive on the temporary Pod
++
+[source,sh]
+----
+kubectl exec ems-data-setup -- unzip /usr/share/data/planet.zip -d /usr/share/planet
+----
+. Delete the temporary Pod and remount the volume into the {ems} Pods as shown <<{p}-maps-pod-configuration,below>>.
++
+[source,sh]
+----
+kubectl delete pod ems-data-setup
+----
+
+[id="{p}-maps-pod-configuration"]
+=== Pod configuration
+You can <<{p}-customize-pods,customize the {ems} Pod>> using a Pod template.
+
+The following example demonstrates how to create a {ems} deployment which mounts a data volume with the complete basemap.
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: maps.k8s.elastic.co/v1alpha1
+kind: ElasticMapsServer
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  elasticsearchRef:
+    name: "quickstart"
+  count: 1
+  podTemplate:
+    spec:
+      containers:
+      - name: maps
+        volumeMounts:
+        - name: map-data
+          readOnly: true
+          mountPath: /usr/src/app/data
+      volumes:
+        - name: map-data
+          persistentVolumeClaim:
+            claimName: ems-basemap
+
+----
+
+The name of the container in the Pod template must be `maps`.
+
+[id="{p}-maps-advanced-configuration"]
+== Advanced configuration
+
+If you already looked at the <<{p}-elasticsearch-specification,Elasticsearch on ECK>> documentation, some of these concepts might sound familiar to you.
+The resource definitions in ECK share the same philosophy when you want to:
+
+* Customize the Pod configuration
+* Customize the product configuration
+* Manage HTTP settings
+
+
+
+[id="{p}-maps-configuration"]
+=== {ems} configuration
+You can add any valid {ems} setting as documented on the link:https://www.elastic.co/guide/en/kibana/current/maps-connect-to-ems.html#elastic-maps-server-configuration[product] page to the `spec.config` section.
```

Comment:
Should we also document `spec.configRef` ?

Reply:
Yes, good point. Will add it.